src/main/java/org/jabref/gui/WaitForSaveFinishedDialog.java

```diff
@@ -0,0 +1,43 @@
+package org.jabref.gui;
+
+import java.util.List;
+
+import javafx.concurrent.Task;
+
+import org.jabref.logic.l10n.Localization;
+
+/**
+ * Dialog shown when closing of application needs to wait for a save operation to finish.
+ */
+public class WaitForSaveFinishedDialog {
+
+    private final DialogService dialogService;
+
+    public WaitForSaveFinishedDialog(DialogService dialogService) {
+        this.dialogService = dialogService;
+    }
+
+    public void showAndWait(List<BasePanel> basePanels) {
+        if (basePanels.stream().anyMatch(BasePanel::isSaving)) {
+            Task<Void> waitForSaveFinished = new Task<Void>() {
+                @Override
+                protected Void call() throws Exception {
```

Comment:
You should ideally use the udpateProgress(ProgressBar.INDETERMINATE_PROGRESS) somewhere at the beginning to make the progress bar moving